dust-lambda-extension/extensions/dust

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env bash
+##
+## Copyright (c) 2025, Crash Override, Inc.
+##
+## This file is part of the Chalk project
+## (see https://crashoverride.com/docs/chalk)
+##
+
+set -euo pipefail
+
+# Enable debug mode only if DEBUG environment variable is set
+[[ ${DEBUG:-} == "true" ]] && set -x
+
+# Logger helpers with consistent formatting and timestamps
+log_with_level() {
+	local level="$1"
+	local color="$2"
+	local message="$3"
+	printf "\e[%sm[%s] [%s] %s\e[0m\n" "$color" "$(date -Iseconds)" "$level" "$message" >&2
+}
+
+LOG_DEBUG() { if [[ ${DEBUG:-} == "true" ]]; then log_with_level "DEBUG" "0;34" "$1"; else true; fi; }
+LOG_INFO() { log_with_level "INFO" "0;32" "$1"; }
+LOG_ERROR() { log_with_level "ERROR" "0;31" "$1"; }
+LOG_WARNING() { log_with_level "WARN" "0;33" "$1"; }
+
+# Configurables (set via environment variables)
+readonly EXTENSION_NAME="${EXTENSION_NAME:-dust}"
+readonly MAX_FAILURES=${MAX_FAILURES:-5}
+
+# Constants
+readonly CHALK_BIN_PATH="$TASK_PATH/chalk"
+readonly CHALK_MARK_PATH="$TASK_PATH/chalk.json"
+readonly TASK_PATH="/var/task"
+readonly TMP_CHALK_PATH="/tmp/chalk"
+
+# Validate critical environment variable
+if [[ -z ${AWS_LAMBDA_RUNTIME_API:-} ]]; then
+	LOG_ERROR "AWS_LAMBDA_RUNTIME_API environment variable is not set"
+	exit 1
+fi
+
+readonly AWS_LAMBDA_RUNTIME_API
+readonly REGISTER_API="http://${AWS_LAMBDA_RUNTIME_API}/2020-01-01/extension/register"
+readonly NEXT_API="http://${AWS_LAMBDA_RUNTIME_API}/2020-01-01/extension/event/next"
+readonly EXIT_API="http://${AWS_LAMBDA_RUNTIME_API}/2020-01-01/extension/exit"
+
+# Global variables
+EXTENSION_ID=""
+EXIT_REQUESTED=false
+
+# Improved error handling for curl requests
+make_api_request() {
+	local method="$1"
+	local url="$2"
+	local headers=("${@:3}")
+
+	local temp_file
+	temp_file=$(mktemp)
+	local http_code
+
+	# Use array to properly handle headers with spaces
+	local curl_args=(-s -w "%{http_code}" -o "$temp_file" -X "$method" "$url")
+	for header in "${headers[@]}"; do
+		curl_args+=(-H "$header")
+	done
+
+	http_code=$(curl "${curl_args[@]}")
+	local curl_exit_code=$?
+
+	if [[ $curl_exit_code -ne 0 ]]; then
+		LOG_ERROR "curl failed with exit code $curl_exit_code for $method $url"
+		rm -f "$temp_file"
+		return 1
+	fi
+
+	if [[ $http_code -lt 200 || $http_code -ge 300 ]]; then
+		LOG_ERROR "HTTP request failed with status $http_code for $method $url"
+		LOG_DEBUG "Response: $(cat "$temp_file")"
+		rm -f "$temp_file"
+		return 1
+	fi
+
+	cat "$temp_file"
+	rm -f "$temp_file"
+	return 0
```

Comment:
due to `-e` in the overall script, if anything fails the temp files wont be cleaned up

in chalk setup action this is handled gracefully:

https://github.com/crashappsec/setup-chalk-action/blob/44e40964dfd031b5d6b77ce5015a418c673ff649/setup.sh#L51-L65